### ISSUE TITLE
audio input

### ISSUE THREAD

**Issue by felixroos** [tidalcycles/strudel#610]
it would be nice if strudel could receive and process audio. I managed to get my audio interface channel 1 like this:

```js
input(0)
```

see https://strudel.tidalcycles.org/?ztG6mAho6aOU

I'm not sure if / how this could be patterned.. though it might be a good start to just support adding audio input.
This is handy if you want to pattern an external instrument via midi and receive back the audio without extra software.
Maybe there is a way to implement stuff like

```js
input(0).room("<0 .5>").lpf(2000)
```

even more maybe, the input could be registered via registerSound, to enable doing:

```js
s("input:0").room("<0 .5>").lpf(2000)
```

This could be tricky, because plugging inputs on and off might create silence or even cracks.

**Comment by felixroos** [comment]
yep, it's complicated: http://localhost:3000/strudel/?dnA4DP1ceQm5

**Comment by felixroos** [comment]
related: https://club.tidalcycles.org/t/baked-in-support-for-midi-in-audio-in-and-audio-devices-channels/5098/4

**Comment by felixroos** [comment]
audio recording example: https://discord.com/channels/779427371270275082/937365093082079272/1221457283968405504 / https://strudel.cc/?jN5kb_k1C3KR

**Comment by ilesinge** [comment]
Improved version of the looper: https://strudel.cc/?uxBZTiojsEW8
